app/src/main/java/org/mifos/mobile/viewModels/ReviewLoanApplicationViewModel.kt

```diff
@@ -63,14 +67,24 @@ class ReviewLoanApplicationViewModel @Inject constructor(var dataManager: DataMa
 
     fun getLoanState() = loanState
 
-    fun submitLoan(): Observable<ResponseBody?>? {
+    private val _submitLoanUiState = MutableLiveData<SubmitLoanUiState>()
```

Comment:
use state flow instead of live data and then make necessary changes in the fragment